packages/react-next/src/components/Pivot/useOverflow.ts

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import { useAsync } from '@uifabric/react-hooks';
+import { getWindow } from '@uifabric/utilities';
+
+export type OverflowCallbacks = {
+  /**
+   * Change whether the item is in the overflow.
+   * This should alter the style of the element to be 'display: none' if it's in the overflow
+   */
+  setItemIsOverflowing: (item: HTMLElement, isOverflowing: boolean) => void;
+
+  /**
+   * Optional: return true if the item is "pinned" and should not be moved to the overflow
+   */
+  isPinned?: (item: HTMLElement) => boolean;
+
+  /**
+   * Notify that the items in the overflow have changed.
+   * This should ensure that the overflow menu is visible, and contains all of the overflowing items.
+   *
+   * @param overflowIndex The index of the first item in the overflow; undefined if there are no items in the overflow
+   */
+  onOverflowItemsChanged?: (overflowIndex: number | undefined) => void;
+};
+
+/**
+ * Track whether any items don't fit within their container, and move them to the overflow menu.
+ *
+ * The overflow menu button must be the last sibling of all of the items that can be put into the overflow:
+ *
+ * <Container>
+ *  <Item /> // Index 0
+ *  <Item /> // Index 1
+ *  ...
+ *  <OverflowMenuButton />
+ * </Container>
+ *
+ * Items will be moved into the overflow from back to front. Any item that is "pinned" (the isPinned callback returns
+ * true) will not be put into the overflow.
+ *
+ * @param overflowMenuButtonRef The DOM element of the overflow menu button
+ * @param callbacks Callback functions to the overflow user, to determine behavior of the overflow
+ */
+export const useOverflow = (
+  overflowMenuButtonRef: React.RefObject<HTMLElement | undefined>,
+  callbacks: OverflowCallbacks,
+) => {
+  let overflowIndex: number | undefined = undefined;
+
+  // Cache the computed styles from this render
+  let cachedStyle: { rtl: boolean; containerPadding: number; menuMargin: number } | undefined = undefined;
+  const getCachedStyle = (container: HTMLElement, menuButton: HTMLElement, win: Window) => {

```

Comment:
use `React.useMemo`

Reply:
useMemo doesn't really work here; the values need to be calculated once the refs have resolved, so it needs to be done during useLayoutEffect. I've updated useOverflow to cache some of the element's size and style info whenever useLayoutEffect runs. If I can improve the deps for useLayoutEffect, it'll end up being similar to useMemo.